pyirf/interpolation/component_estimators.py

```diff
@@ -212,39 +217,61 @@ def __init__(
             grid_points,
         )
 
-        if not isinstance(bin_edges, np.ndarray):
-            raise TypeError("Input bin_edges is not a numpy array.")
-        elif not isinstance(bin_contents, np.ndarray):
+        # Ignore bin_edges if both inter- and extrapolator are NearestNeighborSeacher
+        # or None (for the extrapolator)
+        if issubclass(interpolator_cls, NearestNeighborSearcher):
+            if extrapolator_cls is None:
+                ignore_edges = True
+            elif issubclass(extrapolator_cls, NearestNeighborSearcher):
+                ignore_edges = True
+            else:
+                ignore_edges = False
+        else:
+            ignore_edges = False
+
+        if not isinstance(bin_contents, np.ndarray):
             raise TypeError("Input bin_contents is not a numpy array.")
-        elif bin_contents.shape[-1] != (bin_edges.shape[0] - 1):
-            raise ValueError(
-                f"Shape missmatch, bin_edges ({bin_edges.shape[0] - 1} bins) "
-                f"and bin_contents ({bin_contents.shape[-1]} bins) not matching."
-            )
         elif self.n_points != bin_contents.shape[0]:
             raise ValueError(
                 f"Shape missmatch, number of grid_points ({self.n_points}) and "
                 f"number of histograms in bin_contents ({bin_contents.shape[0]}) "
                 "not matching."
             )
+        if not ignore_edges:
+            if not isinstance(bin_edges, np.ndarray):
+                raise TypeError("Input bin_edges is not a numpy array.")
+            elif bin_contents.shape[-1] != (bin_edges.shape[0] - 1):
+                raise ValueError(
+                    f"Shape missmatch, bin_edges ({bin_edges.shape[0] - 1} bins) "
+                    f"and bin_contents ({bin_contents.shape[-1]} bins) not matching."
+                )
 
         if interpolator_kwargs is None:
             interpolator_kwargs = {}
 
         if extrapolator_kwargs is None:
             extrapolator_kwargs = {}
 
-        if not issubclass(interpolator_cls, DiscretePDFInterpolator):
+        if issubclass(interpolator_cls, DiscretePDFInterpolator):
+            self.interpolator = interpolator_cls(
+                grid_points, bin_edges, bin_contents, **interpolator_kwargs
+            )
+        elif issubclass(interpolator_cls, NearestNeighborSearcher):
```

Comment:
I'd rather also give the `NearestNeighborSearcher` the bin_edges and ignore them there so that all these estimators have the same api then have this class handle different cases depending on the interpolator.

The argument for this structure was that people can implement their own inter/extra polators, but now, some extrapolators need special treatment here?

Reply:
But the user *has* to give something to the component estimator, right? It is just ignored when the estimator is the nearest neighbor finder.

Reply:
Agreed, although I'd rather not call them "Interpolator" nor would I distinct between Inter- and Extrapolators in this very special case, as this is not really any of these.